lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1094,7 +1098,7 @@ sub _create_bam_index {
     return 1;
 }
 
-sub _create_bam_flagstat {
+sub create_bam_flagstat {
```

Comment:
I think it would make sense to check whether or not the flagstat file exists.  That way we needn't lock if it's there and needn't generate it if it has appeared while we wait on the lock. (much like for the header).  In general if the flagstat exists we might not need the `$bam_file` to exist, so we could avoid dying as well!